hive3/src/main/java/org/apache/iceberg/mr/hive/serde/objectinspector/IcebergDateObjectInspectorHive3.java

```diff
@@ -56,7 +56,15 @@ public DateWritableV2 getPrimitiveWritableObject(Object o) {
 
   @Override
   public Object copyObject(Object o) {
-    return o == null ? null : new Date((Date) o);
+    if (o == null) {
+      return null;
+    }
+
+    if (o instanceof Date) {
+      return new Date((Date) o);
+    } else {
+      return o;
+    }
```

Comment:
Why would this be called to copy a non-Date object?

Reply:
On Tez, this is called with LocalDate. On MR we get Date, which is just a wrapper around LocalDate.

Reply:
Why not support both representations that are possibly passed in by Hive?

Reply:
Nevermind, I see that you did.